### PR TITLE
feat(analytics): Project Issue Search Analytics

### PR DESCRIPTION
**Pull request by ehfeng**
Adding analytics for improving project issue search.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx [hunk 1]
```diff
@@ -56,7 +57,7 @@ class EventOrGroupHeader extends React.Component {
     if (includeLink) {
       props.to = `/${orgId}/${projectId}/issues/${isEvent ? groupID : id}/${isEvent
         ? `events/${data.id}/`
-        : ''}`;
+        : ''}${this.props.query ? `?query=${this.props.query}` : ''}`;
```
Comment (ehfeng) [inline src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx]: @billyvg is this the right way to do this? Feels odd.

Comment (billyvg) [reply src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx]: Technically it works - it's very difficult to read though

You could try to define `to` as an object like this: https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/api/Link.md

File: src/sentry/static/sentry/app/views/stream/stream.jsx [hunk 2]
```diff
@@ -476,6 +477,18 @@ const Stream = createReactClass({
         this.transitionTo
       );
     }
+
+    // Ignore saved searches
+    if (this.state.savedSearchList.map(s => s.query == this.state.query).length > 0) {
+      let {orgId, projectId} = this.props.params;
+      HookStore.get('analytics:event').forEach(cb =>
+        cb('issue.search', {
+          query: this.state.queruy,
```
Comment (denamwangi) [inline src/sentry/static/sentry/app/views/stream/stream.jsx]: `this.state.query`

File: src/sentry/static/sentry/app/views/stream/stream.jsx [hunk 3]
```diff
@@ -476,6 +477,18 @@ const Stream = createReactClass({
         this.transitionTo
       );
     }
+
+    // Ignore saved searches
+    if (this.state.savedSearchList.map(s => s.query == this.state.query).length > 0) {
```
Comment (denamwangi) [inline src/sentry/static/sentry/app/views/stream/stream.jsx]: How come we need to go through this map? Instead of grabbing `this.state.query` directly?

Comment (ehfeng) [reply src/sentry/static/sentry/app/views/stream/stream.jsx]: To me, issue search means "user typed in a query and pressed search." Loading a Saved Search (effectively a dashboard) feels less like running a search and more like a pageview (which is why today, it's caught as such).

File: src/sentry/static/sentry/app/views/stream/stream.jsx (hunk 2 above)
Comment (ehfeng) [reply src/sentry/static/sentry/app/views/stream/stream.jsx]: Fixed

File: src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx (hunk 1 above)
Comment (ehfeng) [reply src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx]: Does this look right? https://github.com/getsentry/sentry/pull/7538/commits/3898a2e8d36623851cdc389e6f59c485872fae3c

Comment (billyvg) [reply src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx]: Yeah looks right

**Comment by denamwangi** [comment]
Cool lgtm - might need to add a test in there for that codecov target

**Review by denamwangi — APPROVED** [review]

File: src/sentry/static/sentry/app/views/stream/stream.jsx (hunk 3 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/stream/stream.jsx]: @ehfeng I don't get it, what's the point of map if you just want the length of the array?

Comment (ehfeng) [reply src/sentry/static/sentry/app/views/stream/stream.jsx]: @lynnagara oh, as opposed to doing...a filter? or match? I could switch it to those as well—I can switch to that if you want.

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/stream/stream.jsx]: Do you want to check if the query matches any saved search? I'd use `Array.find` for that. It'll always be true right now unless they've gone and deleted every single saved search. Is it important to filter these out in any case? Loading a saved search won't trigger this function anyway.

**Pull request merged**